src/localstream.js

```diff
@@ -73,8 +73,40 @@ const localstream = (function() {
 	 * @description Flag indicating if the audio and video initialisation is in progress
 	 */
 	let initAudioVideoProgress = false;
+	/**
+	 * @description videoSources array indicating the avaliable video sources
+	 */
+	const videoSources=[];
+	/**
+	 * @description currentVideoSource index indicating the currently used video sources
+	 */
+	let currentVideoSource=0;
+	/**
+	 *@description An array containing audio and video listeners as functions. These listeners will be triggered in case of initiating video/audio by switching camera.
+	 */
+	const streamListenersAudioVideo=[];
 
 	/**
+	 * Call back method for enumerateDevices 
+	 * that is used to get the sources attached to the currently used device/browser
+	*/
+	function gotSources(sourceInfos) {
+		for (let i = 0; i < sourceInfos.length; i++) {
```

Comment:
Use [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead of for:
`videoSources.concat(sourceInfos.filter(sourceInfo => /^videoinput$/.test(sourceInfo.kind)));`